contracts/debt/IncurDebtV1.sol

```diff
@@ -0,0 +1,335 @@
+error IncurDebtV1_AmountAboveBorrowerBalance(uint256 _amount);
+error IncurDebtV1_AmountMoreThanBorrowersLimit(uint256 _borrower);
+error IncurDebtV1_OHMAmountMoreThanAvailableLoan(uint256 _amount);
+error IncurDebtV1_BorrowerHasNoOutstandingDebet(address _borrower);
+error IncurDebtV1_BorrowerStillHasOutstandingDebet(address _borrower);
+
+contract IncurDebtV1 is OlympusAccessControlledV2 {
+    using SafeERC20 for IERC20;
+
+    uint256 public globalDebtLimit;
+    uint256 public totalOutstandingDebt;
+
+    address public immutable OHM;
+    address public immutable gOHM;
+    address public immutable sOHM;
+    address public immutable staking;
+    address public immutable treasury;
+
+    struct Borrower {
+        uint128 debt;
+        uint128 limit;
+        uint128 collateralInSOHM;
+        uint128 collateralInGOHM;
+        bool isAllowed;
+    }
+
+    mapping(address => Borrower) public borrowers;
+
+    constructor(
+        address _OHM,
+        address _gOHM,
+        address _sOHM,
+        address _staking,
+        address _treasury,
+        address _olympusAuthority
+    ) OlympusAccessControlledV2(IOlympusAuthority(_olympusAuthority)) {
+        OHM = _OHM;
+        gOHM = _gOHM;
+        sOHM = _sOHM;
+        staking = _staking;
+        treasury = _treasury;
+    }
+
+    modifier isBorrower(address _borrower) {
+        if (!borrowers[_borrower].isAllowed) revert IncurDebtV1_NotBorrower(_borrower);
+        _;
+    }
+
+    /**
+     * @notice sets the maximum debt limit for the system
+     * - onlyOwner (or governance)
+     * - must be greater than or equal to existing debt
+     * @param _limit in OHM
+     */
+    function setGlobalDebtLimit(uint256 _limit) external onlyGovernor {
+        if (_limit < totalOutstandingDebt) revert IncurDebtV1_LimitBelowOutstandingDebt(_limit);
+        globalDebtLimit = _limit;
+    }
+
+    /**
+     * @notice lets a user become a borrower
+     * - onlyOwner (or governance)
+     * - user must not be borrower
+     * @param _borrower the address that will interact with contract
+     */
+    function allowBorrower(address _borrower) external onlyGovernor {
+        if (borrowers[_borrower].isAllowed) revert IncurDebtV1_AlreadyBorrower(_borrower);
+        borrowers[_borrower].isAllowed = true;
+    }
+
+    /**
+     * @notice sets the maximum debt limit for a borrower
+     * - onlyOwner (or governance)
+     * - limit must be greater than or equal to borrower's outstanding debt
+     * - limit must be less than or equal to the global debt limit
+     * @param _borrower the address that will interact with contract
+     * @param _limit borrower's debt limit in OHM
+     */
+    function setBorrowerDebtLimit(address _borrower, uint256 _limit) external onlyGovernor isBorrower(_borrower) {
+        if (_limit < borrowers[_borrower].debt) revert IncurDebtV1_AboveBorrowersDebtLimit(_limit);
+        if (_limit > globalDebtLimit) revert IncurDebtV1_AboveGlobalDebtLimit(_limit);
```

Comment:
Here we are using globalDebtLimit as raw number of ohm. 

I had a read of this doc today. 

https://www.notion.so/olympusdao/Requirements-Accelerated-Liquidity-3bd9dbc297c14e3498d5e596576c7a39

and it said "Set a global % of circulating OHM as the maximum allowed OHM to be minted for borrow".

Now i think it seems easier to just do what you did with raw number instead of percentage of Ohm supply. I don't mind if %ohm supply keeps going up but if it goes down using it will be problematic. i'l ask on the discord as well to doublecheck.

Reply:
Probably just stick with this just wanted to point this out incase someone has a good reason for it.

Reply:
Its original intention was so that debt could grow without Policy intervention. I don't think it's important for this version of the feature. However, you're right in that we would need to consider supply decreasing. In that case I think existing positions would be grandfathered in and borrow increases disallowed until collateral catches up.